step-controller/step-controller-server/src/main/java/step/migration/tasks/FixPostgreSQLIndexes.java

```diff
@@ -0,0 +1,53 @@
+package step.migration.tasks;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import step.core.Version;
+import step.core.collections.Collection;
+import step.core.collections.CollectionFactory;
+import step.core.collections.Document;
+import step.core.collections.postgresql.PostgreSQLCollection;
+import step.migration.MigrationContext;
+import step.migration.MigrationTask;
+
+public class FixPostgreSQLIndexes extends MigrationTask {
+
+    private static final Logger logger = LoggerFactory.getLogger(FixPostgreSQLIndexes.class);
+    public static final String TIME_SERIES_COLLECTION_PROPERTY = "timeseries";
+    public static final String MEASUREMENTS_COLLECTION_PROPERTY = "measurements";
+    public static final String FUNCTIONS_COLLECTION_PROPERTY = "functions";
+    public static final String PLANS_COLLECTION_PROPERTY = "plans";
+    public static final String TASKS_COLLECTION_PROPERTY = "tasks";
+
+    public FixPostgreSQLIndexes(CollectionFactory collectionFactory, MigrationContext migrationContext) {
+        super(new Version(3,24,3), collectionFactory, migrationContext);
+    }
+
+    @Override
+    public void runUpgradeScript() {
```

Comment:
Do I understand correctly that the required new index are automatically created when needed? (and can not be accidentally re-deleted by this task)?

Reply:
The migration tasks is only executed once (first time started with 3.24.3, the creation of indexes is happening later in the startup lifecyle. We always try to create the index at startup if they do not exist already